VS_Meadow_Extension/VS_Meadow_Extension.Shared/MeadowConnection.cs

```diff
@@ -0,0 +1,51 @@
+﻿using Meadow.CLI;
+using Meadow.Hcom;
+using System;
+using System.Threading;
+
+namespace Meadow
+{
+    static class MeadowConnection
+    {
+        static readonly int RETRY_COUNT = 10;
+        static readonly int RETRY_DELAY = 500;
+
+        internal static IMeadowConnection GetCurrentConnection()
+        {
+            var route = new SettingsManager().GetSetting(SettingsManager.PublicSettings.Route);
```

Comment:
Should keep the SettingsManager.PublicSettings.Route update separate to the extensions, so that selecting a device in the IDE doesn't affect the CLI settings?? But we can standardise on this if you feel strongly about it.